File: backend/app/schemas/ml.py
from pydantic import BaseModel
from typing import Optional

class YieldPredictRequest(BaseModel):
    crop_name: str
    area_acres: float
    soil_type: str
    irrigation_type: str
    season: str
    state: str
    avg_rainfall: Optional[float] = None
    avg_temp: Optional[float] = None

class YieldPredictResponse(BaseModel):
    predicted_yield: float
    confidence_min: float
    confidence_max: float
    unit: str = "quintals"
    crop_name: str
    area_acres: float

class FairPriceRequest(BaseModel):
    crop_name: str
    market_name: str
    state: str
    offered_price: float

class FairPriceResponse(BaseModel):
    crop_name: str
    offered_price: float
    modal_price: float
    deviation_percent: float
    is_anomaly: bool
    severity: str
    anomaly_score: float
    recommendation: str

File: frontend/src/api/ml.ts
import client from "./client";

export const mlAPI = {
  predictYield: (data: {
    crop_name: string;
    area_acres: number;
    soil_type: string;
    irrigation_type: string;
    season: string;
    state: string;
    avg_rainfall?: number;
    avg_temp?: number;
  }) => client.post("/ml/predict-yield", data).then(r => r.data),

  detectAnomaly: (data: {
    crop_name: string;
    market_name: string;
    state: string;
    offered_price: number;
  }) => client.post("/ml/fair-price", data).then(r => r.data),
};
